jmxtrans-output/jmxtrans-output-core/src/main/java/com/googlecode/jmxtrans/model/output/support/opentsdb/OpenTSDBMessageFormatter.java

```diff
@@ -193,7 +206,7 @@ protected void processOneMetric(List<String> resultStrings, Result result, Objec
 				addTag(resultString, addTagName, addTagValue);
 			}
 
-			if (!typeNames.isEmpty()) {
+			if ((typeNames != null) && (!typeNames.isEmpty())) {
```

Comment:
`typeNames` is tagged `@NonNull` in the constructor arguments. We could be more explicit by tagging it `@NonNull` also in the class variables. But in the end, the result is the same, `typeNames` should never be null, so the null check is not needed. If there is a code path where this is not the case, that code path should be fixed.